src/FieldWarning/Assets/UI/Ingame/SelectionManager.cs

```diff
@@ -245,18 +254,49 @@ private bool IsInside(Vector3 t)
 
         private void UnselectAll(List<PlatoonBehaviour> selectedPlatoons, bool justPreviewing)
         {
+            foreach (var wp in _selectedWaypointOverlays)
+            {
+                if (wp != null)
```

Comment:
Is it actually possible for this to be null? If not, I would leave the check out. I think the overlay is ever destroyed but still in the list, we have some bug that we need to fix.